manifests/broker.pp

```diff
@@ -115,41 +115,44 @@
+  String[1] $kafka_version                      = $kafka::params::kafka_version,
+  String[1] $scala_version                      = $kafka::params::scala_version,
+  Stdlib::Absolutepath $install_dir             = $kafka::params::install_dir,
+  Stdlib::HTTPUrl $mirror_url                   = $kafka::params::mirror_url,
+  Boolean $manage_java                          = $kafka::params::manage_java,
+  Stdlib::Absolutepath $package_dir             = $kafka::params::package_dir,
+  Optional[String[1]] $package_name             = $kafka::params::package_name,
+  String[1] $package_ensure                     = $kafka::params::package_ensure,
+  String[1] $user_name                          = $kafka::params::user_name,
+  Stdlib::Absolutepath $user_shell              = $kafka::params::user_shell,
+  String[1] $group_name                         = $kafka::params::group_name,
+  Optional[Integer] $user_id                    = $kafka::params::user_id,
+  Optional[Integer] $group_id                   = $kafka::params::group_id,
+  Boolean $manage_user                          = $kafka::params::manage_user,
+  Boolean $manage_group                         = $kafka::params::manage_group,
+  Stdlib::Filemode $config_mode                 = $kafka::params::config_mode,
+  Stdlib::Absolutepath $config_dir              = $kafka::params::config_dir,
+  Stdlib::Absolutepath $log_dir                 = $kafka::params::log_dir,
+  Stdlib::Absolutepath $bin_dir                 = $kafka::params::bin_dir,
+  String[1] $service_name                       = 'kafka',
+  Boolean $manage_service                       = $kafka::params::manage_service,
+  Enum['running', 'stopped'] $service_ensure    = $kafka::params::service_ensure,
+  Boolean $service_restart                      = $kafka::params::service_restart,
+  Array[String[1]] $service_requires            = $kafka::params::service_requires,
+  Optional[String[1]] $limit_nofile             = $kafka::params::limit_nofile,
+  Optional[String[1]] $limit_core               = $kafka::params::limit_core,
+  Optional[String[1]] $timeout_stop             = $kafka::params::timeout_stop,
+  Boolean $exec_stop                            = $kafka::params::exec_stop,
+  Boolean $daemon_start                         = $kafka::params::daemon_start,
+  Hash $env                                     = {},
+  Hash[String[1], Any] $config                  = {},
+  String[1] $heap_opts                          = $kafka::params::broker_heap_opts,
+  String[1] $jmx_opts                           = $kafka::params::broker_jmx_opts,
+  String[1] $log4j_opts                         = $kafka::params::broker_log4j_opts,
+  String[0] $opts                               = $kafka::params::broker_opts,
+  Boolean $manage_log4j                         = $kafka::params::manage_log4j,
+  Pattern[/[1-9][0-9]*[KMG]B/] $log_file_size   = $kafka::params::log_file_size,
+  Integer $log_file_count                       = $kafka::params::log_file_count
```

Comment:
For the integer here and in the other classes, can you enforce the minimal/maximal values?
example to allow 0 as lowest value and 10 as highest:
```suggestion
  Integer[0, 10] $log_file_count                = $kafka::params::log_file_count
```

Reply:
@bastelfreak Updated like `Integer[1, 50] $log_file_count`